packages/wallet/src/services/DelegationTracker/DelegationTracker.ts

```diff
@@ -71,25 +71,13 @@ export const certificateTransactionsWithEpochs = (
     )
   );
 
-const hasDelegationCert = (certificates: Array<Cardano.Certificate> | undefined): boolean => {
-  if (!certificates || certificates.length === 0) return false;
-
-  return certificates.some((cert) => {
-    let hasCert = false;
-
-    switch (cert.__typename) {
-      case Cardano.CertificateType.StakeDelegation:
-      case Cardano.CertificateType.StakeRegistration:
-      case Cardano.CertificateType.StakeDeregistration:
-        hasCert = true;
-        break;
-      default:
-        hasCert = false;
-    }
-
-    return hasCert;
-  });
-};
+const hasDelegationCert = (certificates: Array<Cardano.Certificate> | undefined): boolean =>
```

Comment:
Giving `certificates parameter` a `[]` default value would let to remove the `!!certificates &&` expression.